cmd/virt-handler/virt-handler.go

```diff
@@ -308,12 +308,16 @@ func (app *virtHandlerApp) Run() {
 	defer close(stop)
 	// Currently nodeLabeller only support x86_64
 	var capabilities *api.Capabilities
+	var pdh string
+	var certChain string
 	if virtconfig.IsAMD64(runtime.GOARCH) {
 		nodeLabellerController, err := nodelabeller.NewNodeLabeller(app.clusterConfig, app.virtCli, app.HostOverride, app.namespace)
 		if err != nil {
 			panic(err)
 		}
 		capabilities = nodeLabellerController.HostCapabilities()
+		pdh = nodeLabellerController.SEV.PDH
+		certChain = nodeLabellerController.SEV.CertChain
```

Comment:
> Node labeller's job is to label the nodes. IIUC, in your implementation, this is not what's being done. Instead, the node labeller simply fetches the information from the node and hold it in-memory as part of its struct. This is misleading and not cohesive, that is, it's outside of node-labeller's responsibilities.


Currently node-labeller fetches the info and adds `kubevirt.io/sev` label to the node (which is later used for proper scheduling of a SEV VM):
https://github.com/kubevirt/kubevirt/blob/86f6b7daddeef6ad704c5d190bedf3c52ec4ebf7/pkg/virt-handler/node-labeller/node_labeller.go#L278-L282
Node-labeller also stores the capabilities. The fetched info includes the certificates and those can be considered as part of platform capabilities.

> virt-handler seems to me like a perfect candidate to fetch this data once needed. This way it wouldn't need to reach node-labeller. The initial implementation can be similar to what you have now, that is the handler can simply fetch is by parsing virsh nodesevinfo. In the future, this can also help solving the CA rotation problem you've mentioned in PR description.

Virt-handler cannot call directly `virsh nodesevinfo` since this needs to be done in the `virt-launcher` context. Initially I implemented it differently. Refering to the commented code https://github.com/kubevirt/kubevirt/pull/7197#discussion_r867452483: instead of `virsh nodesevinfo` I was fetching the certificates from `virt-launcher` by calling libvirt API `GetSEVInfo()`.

I was not sure which approach would be better here. I considered that the certs ware sort of node assets and they do not belong to a VM. Therefore I moved the fetching to node-labeller. So this was one point where I had doubts.

Reply:
But now after some meditation I think the initial approach was right. The benefit of calling libvirt API directly on the VM is that indeed it automatically solves the problem with rotation of the certificates (since we will fetch the latest ones). I will need to double check that approach again but now I am in favour of restoring it instead of using the node labeller.

Reply:
@vasiliy-ul do you need this information before scheduling the VM? Otherwise, you are starting the virt-launcher pod as paused, so you should have communication channel with virt-launcher already open. Or am I missing something?

Reply:
@alicefr , right, the communication channel with `virt-launcher` is already available. In such case there is no need to call `virsh` since its possible to call [libvirt API](https://libvirt.org/html/libvirt-libvirt-host.html#virNodeGetSEVInfo) directly from `virt-launcher`. This approach also works fine. So there are just two options here: we can fetch the certs either from `node-labeller` of from `virt-launcher`.

With node-labeller I wanted to spare on  handler <-> launcher communication and also to cache the certs on the node so they could potentially be fetched without even having a VMI. Nevertheless I think I will switch back to the approach with calling the API in `virt-launcher`. There are some potential issues with permissinos but seems I managed to solve those. So it should work fine and this will automatically handle the certs rotation which is a nice benefit.

Reply:
> the major change I think if you rely on libvirt API is that you need the VMi already up and running.

In fact no, `virNodeGetSEVInfo` does not require a running domain. It needs only a connection to libvirt which is available at that time.

> Why did you change the first approach?

Mostly because virt-launcher is not needed to query the certs since they are per-node assets. By using the node-labeller I was mostly trying to reduce the communication chain and simplify the implementation.